Sources/Logging/Destinations/Log+ConsoleLogDestination.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  Log+ConsoleLogDestination.swift
+//  Alicerce
+//
+//  Created by Meik Schutz on 07/04/2017.
+//  Copyright © 2017 Mindera. All rights reserved.
+//
+
+import Foundation
+
+public extension Log {
+
+    public enum ConsoleLogDestinationError: Swift.Error {
+        case itemFormatFailed(Log.Item, Swift.Error)
+    }
+
+    public class ConsoleLogDestination<ItemFormatter: LogItemFormatter>: LogDestination
+    where ItemFormatter.Output == String {
+
+        public typealias OutputClosure = ((Level, String) -> Void)
+        public typealias LogMetadataClosure = ([AnyHashable : Any]) -> (Level, String)
+
+        public let formatter: ItemFormatter
+
+        public let minLevel: Level
+
+        private let output: OutputClosure
+        private let logMetadata: LogMetadataClosure?
+
+        // MARK: - Lifecycle
+
+        public init(formatter: ItemFormatter,
+                    minLevel: Level = .error,
+                    output: @escaping OutputClosure = { (level, message) in print(message) },
+                    logMetadata: LogMetadataClosure? = nil) {
+
+            self.minLevel = minLevel
+            self.formatter = formatter
+            self.output = output
+            self.logMetadata = logMetadata
+        }
+
+        // MARK: - Public methods
+
+        public func write(item: Log.Item, failure: @escaping (Error) -> Void) {
+
+            let formattedLogItem: String
+            do {
+                formattedLogItem = try formatter.format(logItem: item)
+            } catch {
+                return failure(ConsoleLogDestinationError.itemFormatFailed(item, error))
+            }
+
+            guard !formattedLogItem.isEmpty else { return }
+
+            output(item.level, formattedLogItem)
+        }
+
+        public func setMetadata(_ metadata: [AnyHashable : Any], failure: @escaping (Error) -> Void) {
+
+            guard let (level, message) = logMetadata?(metadata), !message.isEmpty else { return }
+
+            output(level, message)
+        }
+
+        public func removeMetadata(forKeys keys: [AnyHashable], failure: @escaping (Error) -> Void) {
```

Comment:
Should we add a comment stating this method does nothing to prevent the consumers from being confused with it?

Reply:
you mean a comment in the body, for instance? (i.e. not documentation, right?)

Reply:
I meant documentation.

Reply:
Will do